manifests/master/simp_auth.pp

```diff
@@ -1,55 +1,48 @@
 # Add SIMP-specific entries to PuppetServer's auth.conf
 #
+# For documentation about *_allow and *_deny, see the puppetserver docs
+# @see https://puppet.com/docs/puppetserver/5.2/config_file_auth.html#allow-allow-unauthenticated-and-deny
+#
+# @param server_distribution Puppet open source or PE
+#
 # @param auth_conf_path
-#   Type:    Stdlib::AbsolutePath
-#   Default: /etc/puppetlabs/puppetserver/conf.d/auth.conf
 #   The location to the puppet master's auth.conf
 #
 # @param legacy_cacerts_all
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to the PKI cacerts from the legacy `pki` module from all hosts
 #
-# @param legacy_mcollective_all
-#   Type:    Boolean
-#   Default: true
-#   If enabled, allow access to the mcollective cacerts from the legacy `pki` module from all hosts
-#
 # @param legacy_pki_keytabs_from_host
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to each host's own kerberos keytabs from the legacy location
 #
 # @param pki_cacerts_all
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to the cacerts from the `pki_files` module from all hosts
-#
-# @param pki_mcollective_all
-#   Type:    Boolean
-#   Default: true
-#   If enabled, allow access to the mcollective PKI from the `pki_files` module from all hosts
+# @param pki_cacerts_all_allow
+# @param pki_cacerts_all_deny
 #
 # @param keydist_from_host
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to each host's own certs from the `pki_files` module
+# @param keydist_from_host_allow
+# @param keydist_from_host_deny
 #
 # @param krb5_keytabs_from_host
-#   Type:    Boolean
-#   Default: true
 #   If enabled, allow access to each host's own kerberos keytabs from the `pki_files` module
+# @param krb5_keytabs_from_host_allow
+# @param krb5_keytabs_from_host_deny
 #
 class pupmod::master::simp_auth (
   Simplib::ServerDistribution $server_distribution          = simplib::lookup('simp_options::puppet::server_distribution', { 'default_value' => 'PC1' } ),
   Stdlib::AbsolutePath        $auth_conf_path               = '/etc/puppetlabs/puppetserver/conf.d/auth.conf',
   Boolean                     $legacy_cacerts_all           = false,
-  Boolean                     $legacy_mcollective_all       = false,
   Boolean                     $legacy_pki_keytabs_from_host = false,
   Boolean                     $pki_cacerts_all              = true,
-  Boolean                     $pki_mcollective_all          = true,
+  NotUndef                    $pki_cacerts_all_allow        = 'certname',
```

Comment:
I don't understand how 'certname' relates to the regex type

Reply:
Per @jeefberkey 100% puppetserver magic. It's irritating because it's really not obvious how it works. You pretty much have to dig into the template and `auth.conf` docs to figure it out but it *is* technically exposed.